src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/BatchParserWrapper.cs

```diff
@@ -82,15 +82,15 @@ private static List<BatchDefinition> ConvertToBatchDefinitionList(string content
                     }
 
                     // Generate the rest batch definitions
-                    for (int index = 1; index < count - 1; index++)
+                    for (int index = 1; index < count - 1 ; index++)
                     {
                         lineDifference = batchInfos[index + 1].startLine - batchInfos[index].startLine;
                         position = ReadLines(reader, lineDifference, endLine);
                         endLine = position.Item1;
                         endColumn = position.Item2;
                         offset = offsets[index];
-                        batchText = content.Substring(offset, batchInfos[index].length);
-                        startLine = batchInfos[index].startLine;
+                        batchText = batchInfos[index].batchText;
+                        startLine = batchInfos[index].startLine + 1; // this is done for first batch but was missing for following batches
```

Comment:
>// this is done for first batch but was missing for following batches [](start = 69, length = 69)

I'd say just put the first comment here about the pos being 0 based

Reply:
Sure. Just wanted clarity since I (rightly) expected this line to cause questions ;)